Add instructions for Android (#506)
eclipse-cyclonedds/cyclonedds: ports/android/README.md

   *Download* next to *Q*. *Accept* the *License Agreement* and click *Next*
   to download the emulator image. When the download is finished, select *Q*,
   click *Next*, then *Finish* to create the *Android Virtual Device (AVD)*.

> As can be read from [this StackOverflow post](2), it is important to NOT
> select a *Google Play* image, as these images do not allow you to gain
> root privileges.

## Building Eclipse Cyclone DDS

[3]: https://developer.android.com/ndk/guides/cmake
[4]: https://developer.android.com/ndk/guides/cmake#variables

The [Android NDK supports CMake](3) via a toolchain file. Build parameters
such as ABI can specified on the command line. For the complete list of
supported variables, consult the [Toolchain Arguments](4) section.

```
$ cd cyclonedds
$ mkdir build.android
$ cd build.android
$ cmake -DCMAKE_TOOLCHAIN_FILE=<path/to/Android/Sdk>/ndk/<version>/build/cmake/android.toolchain.cmake \
        -DANDROID_ABI=x86 -DANDROID_NATIVE_API_LEVEL=28 -DBUILD_SHARED_LIBS=Off ..
$ cmake --build .
```

> Android supports shared libraries. `BUILD_SHARED_LIBS` is disabled solely
> for demonstration purposes.

> ## Setting up a networking bridge

Reviewer: remove network setup or move to fyi section at end

Author: The network setup is actually the most interesting part because it contains the bare minimum of steps to get things working. Makes debugging etc a lot easier, so a people likely want to have this information once they get going. Apart from that it shows that it'll communicate over the network, not just `lo`. I can move it to the end and add a local test step in between.

$ ip tuntap add dev tap0 mode tap user $USER
$ ip link set tap0 master br0
$ ip link set dev tap0 up promisc on
```

 * Launch the emulator with the *tap* device attached
```
$ <path/to/Android/Sdk>/emulator/emulator -avd Pixel_XL_API_29 -netdelay none -netspeed full -qemu -device virtio-net-pci,netdev=eth1 -netdev tap,id=eth1,ifname=tap0,script=no
```

 * Open a shell in the emulator
```
$ <path/to/Android/Sdk>/platform-tools/adb shell
```

 * Execute `su` from the shell to become *root* and configure the interface.
 * Bring up the interface and assign an IP address.
```
# ip link set dev eth1 up
# ip addr add 192.168.178.2/24 dev eth1
```

 * Make normal routing take precedence over anything that google/qemu configured.
```
$ ip rule add from all lookup main pref 99
```

 * You should now be able to ping the host from the emulator and vice versa.

> The [KDE Community Wiki](5) is an excellent source of information on the
> Android emulator. The second-to-last instruction stems from there and was
> vital to establish communication between the emulator and the host.

[5]: https://community.kde.org/KDEConnect/Android_Emulator

## Deploying and running the RoundTrip example

>  * Push the `RoundTripPong` binary and open a shell.

Reviewer: ```suggestion
 * Push the cyclonedds binaries to Android and open a shell.
```

```

> $ <path/to/Android/Sdk>/platform-tools push <path/to/cyclonedds>/build.android/bin/RoundTripPong /data/local/tmp

Reviewer: ```suggestion
$ <path/to/Android/Sdk>/platform-tools/adb push <path/to/cyclonedds>/build.android/bin/* /data/local/tmp
```

$ <path/to/Android/Sdk>/platform-tools shell
```

> The binary must be copied to the local filesystem, not the sd-card, as that
> is mounted with the `noexec` flag and will not allow you to execute any
> executables.

>  * Make the binary executable (from the shell on the device).

Reviewer: ```suggestion
 * Make cyclonedds binaries executable (from the shell on the device).
```

```
$ cd /data/local/tmp

> $ chmod 755 RoundTripPong

Reviewer: ```suggestion
$ chmod 755 *
```

```

>  * Force the guest to use the `eth0` interface.

Reviewer: remove or move to fyi section at end

```
$ export CYCLONEDDS_URI='<CycloneDDS><Domain><General><NetworkInterfaceAddress>eth0</NetworkInterfaceAddress></General></Domain></CycloneDDS>'
```

>  * Execute `$ ./RoundTripPong`

Reviewer: ```suggestion
 * Execute `$ ./ddsperf -D10 pub & ./ddsperf -D10 sub `
```

>  * Build Eclipse Cyclone DDS for your host machine

Reviewer: remove the sections on building for host & setting up networking OR move them to the end as fyi

>  * Force the host to use the `br0` interface

Reviewer: remove or move to fyi section
